primitives/beefy/src/commitment.rs

```diff
@@ -152,6 +156,115 @@ impl<TBlockNumber> SignedCommitment<TBlockNumber> {
 	}
 }
 
+/// Type to be used to denote placement of signatures
+type BitField = Vec<u8>;
+/// Compress 8 bit values into a single u8 Byte
+const CONTAINER_BIT_SIZE: usize = 8;
+
+/// Compressed representation of [`SignedCommitment`], used for encoding efficiency.
+#[derive(Clone, Debug, PartialEq, Eq, Encode, Decode)]
+struct CompactSignedCommitment<TCommitment> {
+	/// The commitment signatures, as is.
+	commitment: TCommitment,
+	/// A `Vec<u8>`, where each bit in a `u8` represents an `Option<Signature>`
+	/// in the uncompressed GRANDPA validators' signatures.
+	signatures_from: BitField,
+	/// Length of the `Vec` containing uncompressed GRANDPA validators' signatures.
+	signatures_len: u32,
```

Comment:
```suggestion
	/// Number of validators in the Validator Set and hence number of significant bits in the [`signatures_from`] collection.
	///
	/// Note this might be smaller than the size of `signatures_compact` in case some signatures are missing.
	validator_set_len: u32,
```